docs/llvm/overview/index.zh.md

```diff
@@ -0,0 +1,294 @@
+# 新手上路
+
+在本节中，我们将简要叙述与编译器有关的相关基本知识与概念，并介绍 LLVM 项目。
+
+> 本节假设您没有任何与编译器相关的背景且假设您已对编程有所了解，可编写简单的 C 代码。
+
+## 什么是编译器
+
+从概念上讲，编译器是一种将一种编程语言转换为另一种语言的**计算机程序**。而我们通常所说的编译器，一般会将高阶编程语言如 C, C++, Fortran 等转换为更底层的汇编语言或者机器代码。
+
+**注意编译器不是编辑器，更不是 IDE。**
+
+## LLVM
+
+> 注意这里讨论的 LLVM 指包含在 LLVM monorepo 中的所有子项目，而不是指传统意义上的 LLVM 核心。
+
+LLVM 是一组**可重用的**编译器工具链。目前它至少包含以下几个子项目：
+
+* bolt
+
+一个链接后的优化器，通过基于采样剖析器（如Linux perf工具）所收集的执行概况来优化应用程序的代码布局，从而实现改进。
+
+* clang
+
+一个与 GCC 兼容的 C 家族（C, C++, Objective C/C++, OpenCL, CUDA）编译器前端，以可观的编译时间与良好的诊断信息而著名。
+
+* libcxx
+
+C++ 标准库的一个新实现，针对 C++11 及以上版本。
+
+
+* llvm
+
+LLVM 项目的核心，包括 LLVM 中间语言，各种优化及其对应的命令行工具。
+
+* mlir
+
+一个用于编写编译器中各类分析与变换及其所需的中间表示的通用框架。
+
+* lld
+
+一个高性能，模块化的跨平台链接器。
+
+## LLVM 的设计理念
+
+LLVM 是模块化的。LLVM 最重要的方面是它被设计为一组库，而不是像 GCC 那样的单一命令行编译器或像 JVM 或 .NET 虚拟机那样的不透明虚拟机。
+
+举个例子，LLVM 优化器提供了大量 pass，它们被编译成一个或多个 `.o` 文件，然后被构建成一系列静态或动态库。这些库提供各种分析和转换功能，并且 pass 之间尽可能保持独立，或者如果它们依赖于其他分析来完成其工作，则明确声明它们与其他 pass 之间的依赖关系。而在 LLVM 优化实际的优化过程中，只有被使用的 pass 会被链接到最终应用程序，而不是所有的。
```

Comment:
```suggestion
举个例子， LLVM 优化器提供了大量 pass，它们被编译成一个或多个 `.o` 文件，然后被构建成一系列静态或动态库。这些库提供各种分析和转换功能，并且 pass 之间尽可能保持独立，或者如果它们依赖于其他分析来完成其工作，则明确声明它们与其他 pass 之间的依赖关系。而在 LLVM 优化实际的优化过程中，只有被使用的 pass 会被链接到最终应用程序，而不是所有的。
```


> 而在 LLVM 优化实际的优化过程中

Duplicating

> 只有被使用的 pass 会被链接到最终应用程序，而不是所有的。

?? Is that true? executable opt links to libLLVM.so / a lot of xxx.so